core/stat/base/leap_array.go

```diff
@@ -159,7 +159,7 @@ type LeapArray struct {
 }
 
 func NewLeapArray(sampleCount uint32, intervalInMs uint32, generator BucketGenerator) (*LeapArray, error) {
-	if intervalInMs%sampleCount != 0 {
+	if sampleCount == 0 || intervalInMs%sampleCount != 0 {
```

Comment:
Checks that are not zero are checked in the calling function `getRuleStatSlidingWindowBucketCount(r *Rule)` .
Is it necessary to check again?